### ISSUE TITLE
New-User doesn't support specifying custom profile, does it?

### ISSUE THREAD

**Issue by pveller** [SitecorePowerShell/Console#484]
Maybe I am missing something but it looks like I can't create a user based on a custom **User Profile** (aka a custom data template with extra fields that will become my user profile's *custom properties*) using `New-User`. Can I? I am looking at this page as a reference:

http://sitecorepowershell.gitbooks.io/sitecore-powershell-extensions/content/appendix/commands/New-User.html

**Comment by pveller** [comment]
I am looking at how `Sitecore.Shell.Applications.Security.CreateNewUser` does it and it comes down to setting a `ProfileItemId` on the `user.Profile` object and then subsequently saving it (the profile). Maybe that's where I start contributing? :) I have a cmdlet in mind for the versioned layouts thingy for SC8. Do you guys have a simple guide to get started with PSE on local machine (like a contributor guide)?

**Comment by michaellwest** [comment]
Hey @pveller, anything can be done in SPE. In this particular case, the brute force way is like this:

```powershell
New-User -Identity michael -FullName "Michael West" -Enabled -Password "b"

$customUserProfileItemId = "{F7216E06-B7B9-49EB-B0CD-4BF026315247}"
$user = Get-User -Id michael -Authenticated
$user.Profile.ProfileItemId = $customUserProfileItemId
```

What I think would be really helpful is an additional parameter called *ProfileType* (or something like it) with autocomplete that shows the list of available profile names (User, Customer, Employee, etc). This field could accept both the name of the template or a guid.

What do you think?

As far as getting started, here on this [page](http://sitecorepowershell.gitbooks.io/sitecore-powershell-extensions/content/installation.html)  that takes you [here](http://blog.najmanowicz.com/2015/03/03/set-up-sitecore-powershell-extensions-development-environment/). The steps are slightly updated and could/should certainly be moved to the gitbook. 

There is a Setup-Folders.ps1 script in the root of the repo and a deploy.targets sample file (under Cognifide.PowerShell).

If you go through this process then perhaps you can help write up the current steps for the book too.

**Comment by pveller** [comment]
Right. I just did exactly same thing. `New` followed by `Get` and then setting the profile.

And yes, I agree that it'd be nice having a `ProfileType` parameter that would do it all behind the scenes.

**Comment by michaellwest** [comment]
Then it's settled. You'll be assigned to this issue. Can we have it by version 3.4? Feel free to submit a pull request.

**Comment by pveller** [comment]
:) sure! what's the schedule for 3.4?

**Comment by NDurham12** [comment]
Good idea. Be sure to call the save command after setting. Had an urgent
prod issue because I forgot to call the save and the profile would reset
after a user changed the password.

**Comment by michaellwest** [comment]
@NDurham12 Did that issue happen while usinig the `Set-User` and `Set-UserPassword` commands?

**Comment by NDurham12** [comment]
No the passwords were reset using normal password reset mvc controller
code. However the accounts were created by using a spe script. In the
script I didn't call the profile.save cmd after adding the profile. Took
quite a lot of dB tracing to find the root cause, since the custom profile
was visible on the user without calling the save.

**Comment by AdamNaj** [comment]
Implemented as #484 - closing this one as duplicate.
